### PR TITLE
Fixing some doc typos and issues

### PR DESCRIPTION
**Pull request by jparrill**
In the documentation website we can see a typo that makes the Markdown to not render properly: https://hypershift-docs.netlify.app/how-to/distribute-hosted-cluster-workloads/. 

Also reviewing the API documentation I've figured out that one of the APIs has not the documentation for oc explain command: `hostedcontrolplanes.hypershift.openshift.io.spec.releaseImage`

Signed-off-by: Juan Manuel Parrilla Madrid <jparrill@redhat.com>

**What this PR does / why we need it**:

- Fixes documentation issues in the API and in the Website

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [X] Subject and description added to both, commit and PR.
- [X] Relevant issues have been referenced.
    - I don't think deserves to create an issue or Jira for that minimal change
- [X] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by sjenning** [comment]
Run `make update` to fix `verify` issues.  Other than that, looks good :+1:

File: api/v1alpha1/hosted_controlplane.go [hunk 1]
```diff
@@ -25,9 +25,12 @@ type HostedControlPlane struct {
 
 // HostedControlPlaneSpec defines the desired state of HostedControlPlane
 type HostedControlPlaneSpec struct {
+	// ReleaseImage is the release image applied to the hosted control plane.
 	ReleaseImage string                      `json:"releaseImage"`
 	PullSecret   corev1.LocalObjectReference `json:"pullSecret"`
-	IssuerURL    string                      `json:"issuerURL"`
+	// IssuerURL is an OIDC issuer URL which is used as the issuer in all
+	// used by the service account token issuer. The secret is expected to contain
```
Comment (csrwng) [inline api/v1alpha1/hosted_controlplane.go:32]: to contain... ?

**Review by csrwng — COMMENTED** [review]
Just a comment. Thx!

In general, the docs for HostedControlPlane are not as relevant as those for HostedCluster and NodePool since those are our external types.

File: api/v1alpha1/hosted_controlplane.go (hunk 1 above)
Comment (jparrill) [reply api/v1alpha1/hosted_controlplane.go]: fixed

File: api/v1alpha1/hosted_controlplane.go [hunk 2]
```diff
@@ -25,9 +25,14 @@ type HostedControlPlane struct {
...
 type HostedControlPlaneSpec struct {
+	// ReleaseImage is the release image applied to the hosted control plane.
 	ReleaseImage string                      `json:"releaseImage"`
 	PullSecret   corev1.LocalObjectReference `json:"pullSecret"`
-	IssuerURL    string                      `json:"issuerURL"`
+	// IssuerURL is an OIDC issuer URL which is used as the issuer in all
+	// used by the service account token issuer. The secret is expected to point to
+	// an S3 bucket which contains the OIDC authentication inside
+	// (E.G https://hypershift-ci-1-oidc.s3.us-east-1.amazonaws.com/sample-dev-6r5rf)
+	IssuerURL string `json:"issuerURL"`
```
Comment (zanetworker) [inline api/v1alpha1/hosted_controlplane.go:35]: Is this field provider specific, i.e., do we always expect to point to s3?

File: cmd/install/assets/hypershift-operator/hypershift.openshift.io_hostedcontrolplanes.yaml [hunk 3]
```diff
@@ -1990,6 +1990,10 @@ spec:
                   The default value is SingleReplica.
                 type: string
               issuerURL:
+                description: IssuerURL is an OIDC issuer URL which is used as the
+                  issuer in all used by the service account token issuer. The secret
```
Comment (zanetworker) [inline cmd/install/assets/hypershift-operator/hypershift.openshift.io_hostedcontrolplanes.yaml:1994]: same comment as above :)

File: api/v1alpha1/hosted_controlplane.go (hunk 2 above)
Comment (jparrill) [reply api/v1alpha1/hosted_controlplane.go:35]: Ohh that was my bad, it's not provider specific. Let me fix it :)

**Review by zanetworker — APPROVED** [review]

**Pull request merged**